lib/desktop/config.rb

```diff
@@ -26,234 +26,142 @@
 # ==============================================================================
 require_relative 'type'
 require 'xdg'
-require 'tty-config'
 require 'fileutils'
 
-module Desktop
-  module Config
-    class << self
-      DESKTOP_DIR_SUFFIX = File.join('flight','desktop')
+require 'flight_configuration'
 
-      def data
-        @data ||= TTY::Config.new.tap do |cfg|
-          cfg.append_path(File.join(root, 'etc'))
-          begin
-            cfg.read
-          rescue TTY::Config::ReadError
-            nil
-          end
-        end
-      end
+module Flight
+  def self.config
+    @config ||= Desktop::Configuration.load
+  end
 
-      def save_data
-        FileUtils.mkdir_p(File.join(root, 'etc'))
-        data.write(force: true)
-      end
+  def self.env
+    @env ||= 'production'
+  end
 
-      def data_writable?
-        File.writable?(File.join(root, 'etc'))
-      end
+  def self.root
+    @root ||= File.expand_path('../..', __dir__)
+  end
+end
 
-      def user_data
-        @user_data ||= TTY::Config.new.tap do |cfg|
-          xdg_config.all.map do |p|
-            File.join(p, DESKTOP_DIR_SUFFIX)
-          end.each(&cfg.method(:append_path))
-          begin
-            cfg.read
-          rescue TTY::Config::ReadError
-            nil
-          end
-        end
-      end
+module Desktop
+  class Configuration
+    extend FlightConfiguration::DSL
 
-      def save_user_data
-        FileUtils.mkdir_p(
-          File.join(
-            xdg_config.home,
-            DESKTOP_DIR_SUFFIX
-          )
-        )
-        user_data.write(force: true)
+    class << self
+      # Override the config files to use the legacy paths
+      def config_files(*_)
+        @config_files ||= [global_config, user_config]
```

Comment:
Previously, the user config could only specify a few values; `geometry`, `desktop_type`, `bg_image`, `session_env_path` and `session_env_override`.  With the change to `flight_configuration` it is now possible for the user config to specify any config value.

How much effort would it be to restrict the user config to just the values that it could be specify previously?

Reply:
I can't see a way to do this immediately without getting `flight_configuration` involved. There is likely scope for "user configurations" within `flight_configuration`.

The best I can thick of is:

* Remove `user_config` from `config_files`, and
* "preload" `user_config` and set the above mentioned keys in the `ENV`.

A bit of care would need to be taken to ensure sub-shells do not inherit the envs. However there is already a `with_cleanest_env` method, which could preform the cleanup.